tiatoolbox/annotation/storage.py

```diff
@@ -1007,6 +1050,195 @@ def select_values(
+                run via pickle or the parsing of the string statement.
+            n_where (str or bytes or Callable):
+                Predicate to filter the nearest annotations by. Defaults
+                to None (assume always true). See `where` for more
+                details.
+            distance (float):
+                The distance to search for annotations within. Defaults to
+                5.0.
+            geometry_predicate (str):
+                The predicate to use when comparing geometries. Defaults
+                to "intersects". Other options include "within" and
+                "contains". Ignored if `mode` is "boxpoint-boxpoint" or
+                "box-box".
+            mode (tuple[str, str] or str):
+                The method to use for determining distance during the
+                query. Defaults to "box-box". This may significantly
+                change performance depending on the backend. Possible
+                options are:
+                  - "poly-poly": Polygon boundary to polygon boundary.
+                  - "boxpoint-boxpoint": Bounding box centre point to
+                    bounding box centre point.
+                  - "box-box": Bounding box to bounding box.
+                May be specified as a dash separated string or a tuple
+                of two strings. The first string is the mode for the
+                query geometry and the second string is the mode for
+                the nearest annotation geometry.
+
+        Returns:
+            Dict[str, Dict[str, Annotation]]:
+                A dictionary mapping annotation keys to another
+                dictionary which represents an annotation key and all
+                annotations within `distance` of it.
+
+        The `mode` argument is used to determine how to calculate the
+        distance between annotations. The default mode is "box-box".
+
+        The "box-box" mode uses the bounding boxes of stored annotations
+        and the query geometry when determining if annotations are
+        within the neighbourhood.
+
+        .. figure:: ../images/nquery-box-box.png
+            :width: 512
+            :alt: "box-box" mode
+
+        The "poly-poly" performs full polygon-polygon intersection with
+        the polygon boundary of stored annotations and the query
+        geometry to determine if annotations are within the
+        neighbourhood.
+
+        .. figure:: ../images/nquery-poly-poly.png
+            :width: 512
+            :alt: "poly-poly" mode
+
+
+        The "boxpoint-boxpoint" mode uses the centre point of the
+        bounding box of stored annotations and the query geometry when
+        determining if annotations are within the neighbourhood.
+
+        .. figure:: ../images/nquery-boxpoint-boxpoint.png
+            :width: 512
+            :alt: "boxpoint-boxpoint" mode
+
+
+        Examples:
+            Example bounding boxy query with one neighbour within a
```

Comment:
Example bounding box query

Reply:
@measty What does this comment mean? Is it a suggested rephrasing?